# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/serviceworkers/ServiceWorkerPrivate.cpp
# Commit: b9d73f961292
# Full Hash: b9d73f9612929dda88ca7e68d8d2e04de2aa2a86
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2023-01-20 21:21:03
# Regressor Bug: 1351231
# File Overlap Count: 2
# Description:
#   Bug 1351231 - FetchService integration for PFetch. r=dom-worker-reviewers,jesup
#   
#   Depends on D142436
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D142437
# ==============================================================================

diff -r 989cbb08b87e -r b9d73f961292 dom/serviceworkers/ServiceWorkerPrivate.cpp
--- a/dom/serviceworkers/ServiceWorkerPrivate.cpp	Fri Jan 20 09:09:17 2023 +0000
+++ b/dom/serviceworkers/ServiceWorkerPrivate.cpp	Fri Jan 20 09:09:17 2023 +0000
@@ -1566,7 +1566,8 @@
     MOZ_ALWAYS_SUCCEEDS(
         aChannel->GetChannel(getter_AddRefs(underlyingChannel)));
     RefPtr<FetchService> fetchService = FetchService::GetInstance();
-    return fetchService->Fetch(std::move(preloadRequest), underlyingChannel);
+    return fetchService->Fetch(AsVariant(FetchService::NavigationPreloadArgs{
+        std::move(preloadRequest), underlyingChannel}));
   }
   return FetchService::NetworkErrorResponse(NS_ERROR_UNEXPECTED);
 }
